modules/database/src/std/softIoc/softMain.cpp

```diff
@@ -109,8 +109,20 @@ void lazy_dbd(const std::string& dbd_file) {
 
     if (verbose)
         std::cout<<"dbLoadDatabase(\""<<dbd_file<<"\")\n";
-    errIf(dbLoadDatabase(dbd_file.c_str(), NULL, NULL),
-          std::string("Failed to load DBD file: ")+dbd_file);
+
+    if (dbLoadDatabase(dbd_file.c_str(), NULL, NULL)){
+        if(verbose)
+            std::cout<<"Failed to load DBD file: "+dbd_file+"\n";
+        std::cout<<"Trying to load relative to EPICS_BASE instead.\n";
+        std::string dbd_base(DBD_FILE);
+        if(dbLoadDatabase(dbd_base.c_str(), NULL, NULL)){
```

Comment:
I had been imagining concatenating the paths and using the existing dbd search path, but I see now that this would be making an assumption about the file name.

I'm also wondering if silently falling back on any failed load will cause confusion if eg. someone runs `softIoc -D /path/with/typo.dbd`.  Seems like this should be an error.  So the fallback would only be if `dbd_file==DBD_FILE`.

As for how to fall back reasonably.  @anjohnson is correct that calling `dbLoadDatabase()` twice is a bad idea.  This will have unfortunate consequences if the first failure was due to a syntax error.  (eg. someone messing with `menuScan`)

It occurs to me though that [`dbOpenFile()`](https://github.com/epics-base/epics-base/commit/62092d4399e67eefee4089892cc1cd29395d79fe#diff-0465e0aca1de11876beea5b8eef174b4ed00bf0d89b66d1f3c2f2b0406fbb671R306-R313) now offers a way to try opening a file w/o parsing it.

So the logic could be something like:

```c++
FILE* fp = NULL;
if(!dbOpenFile(pdbbase, dbd_file.c_str(), &fp) && dbd_file==DBD_FILE) {
    dbOpenFile(pdbbase, DBD_FILE, &fp)
}
if(!fp) { ... error, unable to open file ... }
else if(dbLoadDatabaseFP(&pdbbase, fp, NULL, NULL)) {
   ... syntax error or unable to open include'd ...
```